src/compiler/parser.ts

```diff
@@ -17,19 +17,19 @@ namespace ts {
     }
 
     function visitNode<T>(cbNode: (node: Node) => T, node: Node): T {
-        if (node) {
+        if (node !== void 0) {

```

Comment:
After running some more tests, the difference between implicit casts and `void 0` appears to exist entirely within the imagination of the Chrome profiler. When running normally, it doesn't seem to make any difference.